backend/src/services/authService.js

```diff
@@ -33,11 +33,11 @@ const AuthService = {
   async Login(email, password) {
     const user = await models.User.findOne({ where: { email: email } });
 
-    const userAuth = await models.Auth.findOne({ where: { userId: user.id } });
-
     // No user, return error message
     if (!user) return { error: "Invalid username or password" };
 
+    const userAuth = await models.Auth.findOne({ where: { userId: user.id } });
```

Comment:
bug found while testing, if user isn't defined then the program crashes because it cannot resolve the `user.id`